### ISSUE TITLE
Nightly Compile Time Regression (06/23 to 07/01)

### ISSUE THREAD

**Issue by steffengy** [rust-lang/rust#34630]
I noticed a pretty huge compile time regression from 
`rustc 1.11.0-nightly (ad7fe6521 2016-06-23)` to `rustc 1.11.0-nightly (01411937f 2016-07-01) `.

```diff
- time: 7.904; rss: 120MB	expansion
+ time: 84.063; rss: 121MB	expansion
```

Compiling the same project, results in a pretty big increase in expansion time.
I could notice similar (**about 11x**) slow-downs on my windows installation and
reproduced it on a fresh ubuntu.

**All instructions used can be found here**: https://gist.github.com/steffengy/ab3660ec9d6eecb0fc38e7d536d9e7ef

**Comment by nagisa** [comment]
cc @jseyfried could be your macro-related changes.

**Comment by alexcrichton** [comment]
triage: I-nominated

**Comment by jonas-schievink** [comment]
cc me (want to check this on my assembler macro)

**Comment by Aatch** [comment]
The project in question is [steffengy/pesty-php](https://github.com/steffengy/pesty-php), for reference.

**Comment by jseyfried** [comment]
I'll look into it.

**Comment by steffengy** [comment]
bisect shows that the first bad commit is 5bf7970ac70b4e7781e7b2f3816720aa62fac6fd

**Comment by jseyfried** [comment]
@steffengy Thanks for bisecting -- I fixed this in #34652.
